### DISCUSSION TITLE
Importing split transactions from GnuCash

### DISCUSSION THREAD

**Discussion by radam9** [firefly-iii/firefly-iii#6197] [Firefly III] [answered]
Hey there,
I am trying to import my history from a gnucash csv export, and I have quite a few split transactions in the history.
During the import stage I can map the I can map the columns of the csv with the fields in firefly, but since the split transactions are in consecutive rows in the gnucash csv, the import doesn't work properly.
I was wondering if someone could help me out with the import.
I am ok with transforming the csv to work with the import, but I do not know what the format should be.

A sample of the GnuCash csv can be seen in the screenshot below.
![Screenshot from 2022-06-27 20-49-52](https://user-images.githubusercontent.com/14952219/176014174-a608a204-f915-41c7-a1bb-0bca4f766bd5.png)

As you can see in the screenshot that split transactions are all included under one `transaction id` and are in the rows subsequent to the main transaction row.

PS: is it also possible to import the the initial transaction that imports my capital and sets my opening balance by setting the initial state of each account?

Thanks in advanced :)

**Comment by JC5** [comment, marked as answer]
The data importer can't handle split transactions. There's no way to get those in, I never built support for it.

**Reply by radam9** [reply]
Thanks for the prompt reply.
It is a shame the importer doesn't support that.

would it be too complicated to add the split transactions to the database directly using a script or something?
or are there too many dependencies between the tables in the database or too much logic happening in the background?

**Reply by JC5** [reply]
Skip the database, use the API. If you can convert your splits into a JSON object, you're in:

https://api-docs.firefly-iii.org/#/transactions/storeTransaction

**Reply by radam9** [reply]
> Skip the database, use the API. If you can convert your splits into a JSON object, you're in:
> 
> https://api-docs.firefly-iii.org/#/transactions/storeTransaction

ooh, that looks great!
one last question:
from what I understand firefly works with double entry accounting, but when importing using the API or DataImporter, would I pass both transactions or only 1?

**Reply by JC5** [reply]
Just the one :)

**Reply by radam9** [reply]
great, thanks a lot !
